### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by jschueller**
- #2780 
- #1417 
- MultiStart fixes for overflow in reported number of calls & history @sofianehaddad 
 see http://openturns.github.io/openturns/latest/auto_numerical_methods/optimization/plot_optimization_rastrigin.html
- Fixes for undocumented Normal/Student dimension ctors
- Renamed DistributionChecker to DistributionValidation for consistency with KL/LM/FC/...Validation etc @mbaudin47 
- A workaround for another matplotlib 3.6.x contour bug in the colorbar throwing IndexError

### CODE REVIEW

**Review by sofianehaddad — APPROVED** [review]
Thanks for the `MultiStart`
Ok also for renaming

File: lib/src/Uncertainty/Distribution/Normal.cxx [hunk 1]
```diff
@@ -102,7 +101,6 @@ Normal::Normal(const Point & mean,
                            , CorrelationMatrix(mean.getDimension())
                            , 1.0)
   , logNormalizationFactor_((-1.0 * mean.getDimension()) * SpecFunc::LOGSQRT2PI)
-  , hasIndependentCopula_(false)
```
Comment (josephmure) [inline lib/src/Uncertainty/Distribution/Normal.cxx:105]: Shouldn't you use checkIndependentCopula in this constructor?

**Review by josephmure — APPROVED** [review]
Let's get this into 1.24, but can you please first check my comment above?

**Comment by jschueller** [comment]
I added two other commits to simplify the barplot example and workaround a matplotlib bug in viewer

**Comment by josephmure** [comment]
Maybe update the ChangeLog with DistributionValidation (which can be useful to check a PythonDistribution).

**Comment by jschueller** [comment]
> Maybe update the ChangeLog with DistributionValidation (which can be useful to check a PythonDistribution).

will do, a global update of Changelog is needed

**Pull request merged**